### PR TITLE
adding brands api guide updates

### PR DESCRIPTION
**Pull request by chrisdmills-okta**
## Description:
- **What's changed?** 

As per https://oktainc.atlassian.net/browse/OKTA-413691, we wanted to add to our guide content to explain how to use the new Brands API to update your okta org theme (colors, icon, etc.)

I chose to add a single guide that explains all of what the Brands API can do, and then link to it from all the places where knowing about it would be useful (e.g. the guide to updating email templates, error pages, etc.)

- **Is this PR related to a Monolith release?** Yes. 2021.08.0

### Resolves:

* [OKTA-413691](https://oktainc.atlassian.net/browse/OKTA-413691)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+Once you've set the `themeId` variable to a specific theme ID, you can return a specific theme response object using the following request (**Get theme** in Postman):
+
+<ApiOperation method="get" url="/api/v1/brands/{brandId}/themes/{themeId}" />
+
+This will return a [theme response object](/docs/reference/api/brands/#theme-response-object), which contains all the details of your theme, including logo, favicon, colors, and background image.
+
+## Updating your theme
+
+In this section we'll look at how we can update all of the different facets of your theme.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:95]:
```suggestion
In this section we look at how we can update all of the different facets of your theme.
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+### Update theme colors and asset placement
+
+You can update your theme colors and the settings that dictate where your assets will be used with the following request (**Update theme** in Postman):
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:99]:
```suggestion
You can update your theme colors and the settings that dictate where your assets are used with the following request (**UPDATE theme** in Postman):
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+<ApiOperation method="put" url="/api/v1/brands/{brandId}/themes/{themeId}" />
+
+The request body needs to contain a JSON object along these lines:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:103]:
```suggestion
The request body needs to contain a JSON object:
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  "signInPageTouchPointVariant": "BACKGROUND_IMAGE",
+  "endUserDashboardTouchPointVariant": "FULL_THEME",
+  "errorPageTouchPointVariant": "BACKGROUND_IMAGE",
+  "emailTemplateTouchPointVariant": "FULL_THEME"
+}
+```
+
+  > **Note:** The different properties are explained in the [Theme object reference](/docs/reference/api/brands/#theme-object).
+
+  > **Note:** To see your custom colors and image assets being used on your Okta org, you need to set the `...Variant` properties to non-`OKTA_DEFAULT` values when you update your theme. See [Variant Definition](/docs/reference/api/brands/#variant-definition) for explanations of the available variant values.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:118]:
```suggestion
  > **Note:** To see your custom colors and image assets being used on your Okta org, you need to set the `...Variant` properties to non-`OKTA_DEFAULT` values when you update your theme. See [Variant definition](/docs/reference/api/brands/#variant-definition) for explanations of the available variant values.
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+A successful request will result in an updated [Theme response object](/docs/reference/api/brands/#theme-response-object).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:120]:
```suggestion
A successful request results in an updated [Theme response object](/docs/reference/api/brands/#theme-response-object).
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+### Update and delete theme logos and images
+
+You can upload a logo, favicon, and background image to be used in your theme (on your email templates, your End-User Dashboard, etc.) with the following requests (**Upload logo**, **Upload favicon**, and **Upload background image** in Postman):
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:124]:
```suggestion
You can upload a logo, favicon, and background image for use in your theme (on your email templates, your Okta End-User Dashboard, and so on) with the following requests (**Upload logo**, **Upload favicon**, and **Upload background image** in Postman):
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+<ApiOperation method="post" url="/api/v1/brands/{brandId}/themes/{themeId}/logo" />
+<ApiOperation method="post" url="/api/v1/brands/{brandId}/themes/{themeId}/favicon" />
+<ApiOperation method="post" url="/api/v1/brands/{brandId}/themes/{themeId}/background-image" />
+
+The request body needs to be form-data, with a file property containing a file to upload. Guidelines are as follows:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:130]:
```suggestion
The request body needs to be form data with a file property that contains a file to upload. Guidelines are as follows:
```

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+* Logos must be in PNG, JPG, or GIF format and less than 1 MB in size. The file dimensions should be less than 3840 x 2160 px. For best results use landscape orientation, a transparent background, and a minimum size of 420px by 120px to prevent upscaling.
+* Favicons must be in PNG or ICO format and must be in 1:1 ratio with a maximum dimension of 512 x 512 px.
+* Background images must be a PNG, JPG, or GIF file, and be less than 2MB in size.
+
+A successful response will result in a JSON object containing the `url` of the uploaded image, e.g.:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:136]:
```suggestion
A successful response results in a JSON object that contains the `url` of the uploaded image, such as:
```

File: packages/@okta/vuepress-site/docs/guides/email-customization/before-you-begin/index.md [hunk 9]
```diff
@@ -8,6 +8,10 @@ You can customize and style the default email notifications that Okta sends to e
 
 Email notifications are based on templates that are generated automatically and sent to end users according to your settings. Okta email templates contain text for account activation, password reset, and account unlock scenarios and are available in each Okta-supported language. You can use the default email templates as is, or you can edit the email template text to send end users custom Okta-generated email messages.
 
+## Use the Brands API
+
+The [Brands API](/docs/reference/api/brands/) is a more recent feature (currently in Early Access) that allows you to set icons, images, and colors across your Okta-hosted sign-in widget, error pages, email templates, and End-User Dashboard all at once, without needing to set a customized Okta URL domain. To find out more about this feature and how to use it, see [Customize your Okta experience with the Brands API](/docs/guides/customize-themes).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/email-customization/before-you-begin/index.md:13]:
```suggestion
The [Brands API](/docs/reference/api/brands/) is a feature (currently in Early Access) that allows you to set icons, images, and colors across your Okta-hosted Sign-In Widget, error pages, email templates, and Okta End-User Dashboard all at once, without needing to set a customized Okta URL domain. To find out more about this feature and how to use it, see [Customize your Okta experience with the Brands API](/docs/guides/customize-themes).
```

File: packages/@okta/vuepress-site/docs/guides/style-the-widget/style-okta-hosted/index.md [hunk 10]
```diff
@@ -39,6 +39,10 @@ If you are familiar with using HTML and want to change the page layout, colors,
 
 > **Note:** See the <GuideLink link="../customization-examples">Customization examples</GuideLink> section for examples that you can alter and use on your hosted sign-in page.
 
+### Use the Brands API
+
+The [Brands API](/docs/reference/api/brands/) is a more recent feature (currently in Early Access) that allows you to set icons, images, and colors across your Okta-hosted sign-in widget, error pages, email templates, and End-User Dashboard all at once, without needing to set a customized Okta URL domain. To find out more about this feature and how to use it, see [Customize your Okta experience with the Brands API](/docs/guides/customize-themes).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/style-the-widget/style-okta-hosted/index.md:44]:
```suggestion
The [Brands API](/docs/reference/api/brands/) is a feature (currently in Early Access) that allows you to set icons, images, and colors across your Okta-hosted sign-in widget, error pages, email templates, and End-User Dashboard all at once, without needing to set a customized Okta URL domain. To find out more about this feature and how to use it, see [Customize your Okta experience with the Brands API](/docs/guides/customize-themes).
```

**Review by susanharper-okta — COMMENTED** [review]
couple items and some questions. thanks!

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+It is up to you how you make requests to the API to update your branding; we will be demonstrating the required API calls using a Postman collection to demonstrate them in a language/platform neutral way. To run the API calls:
+
+1. [Create an API token](/docs/guides/create-an-api-token/overview/) to use when accessing the API.
+1. [Download](https://www.postman.com/downloads/) and Install Postman.
+1. Once Postman is installed, import the Okta environment and add your Okta domain and API token to Postman, as explained in [Use Postman with the Okta REST APIs > Set up your environment](/code/rest/#set-up-your-environment).
+1. Click the following button to add the Brands collection to postman, which will allow you to test the API calls described below.
+
+[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/1d58ab8a3909dd6a3cfb)
+
+## Important: overriding themes in the code template editors
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:36]: I wanted to include it as a heading to really draw attention to it - but you are right, we should probably remove the "Important:"

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+You can customize individual parts of your Okta org experience using the various code editors we've made available in the Okta Admin UI (for example **Settings** > **Customizations** > **Custom Sign-in** to customize the Okta-hosted sign-in page. Please note that when you use the Brands API, your custom brand settings will not apply to the places where you've customized the theme settings.
+
+So for example, if you make changes to the sign-in page code using the editor referenced above and change the background image or logo setting, your customizations will override the Brands API settings. To get your Brands API settings back, you will have to reset the code editors to the default code again.
+
+## Top-level overview: Brands and themes
+
+At the top level, Your Okta organization contains a brand, which contains a default theme. The default brand is applied to your organization's subdomain/[custom domain](https://developer.okta.com/docs/guides/custom-url-domain/overview/) if you have specified one.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:44]: Ah, bit of confusion here - I meant specify the subdomain/custom domain, not specify the brand.

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+
+  > **Note:** At the time of writing each org can only contain one brand and one theme, however we are working on a plan to allow multiple themes and multiple brands per org, so stay tuned!
+
+### Get brands
+
+You can return the brands with the following request (**Get brands** in Postman):
+
+<ApiOperation method="get" url="/api/v1/brands" />
+
+This returns an array of [brand response objects](/docs/reference/api/brands/#brand-response-object). You can try this in Postman by running the Get brands request.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:54]: Ah, in this case it should be `**Get brands**` — the name of the request as written in Postman is "Get brands".

[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:54]: So, no, in the case of the names of the Postman request names.

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+
+You can also return a specific brand by running the **Get brand** request. Before you run the request, you'll need to set the `brandId` variable in Postman, which is used in the request, as seen below.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:56]: I agree with removing "you'll need to here", but not the change to "Get"

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+
+  > **Tip:** The easiest way to set a variable in Postman is to highlight the value you want to assign to the variable (for example you can find the ID of a specific brand in the brand response object returned by `GET /api/v1/brands`), right/Ctrl + click on the value, and select **Set: _your-environment-name_ > _your-variable-name_**.
+
+<ApiOperation method="get" url="/api/v1/brands/{brandId}" />
+
+This returns a single brand response object.
+
+### Get brands
+
+You can also update brand information with the following request (**Update brand** in Postman):
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:66]: Again, see above.

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+``` json
+{
+  "agreeToCustomPrivacyPolicy": true,
+  "customPrivacyPolicyUrl": "https://www.someHost.com/privacy-policy"
+}
+```
+
+### Get themes
+
+You can return the themes contained in a brand with the following request (**Get themes** in postman):
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:81]: "Postman" yes, "GET" no, as per above.

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+
+<ApiOperation method="put" url="/api/v1/brands/{brandId}/themes" />
+
+This returns an array of [theme response objects](/docs/reference/api/brands/#theme-response-object).
+
+Once you've set the `themeId` variable to a specific theme ID, you can return a specific theme response object using the following request (**Get theme** in Postman):
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:87]: "After" yes, "GET" no, as per above.

File: packages/@okta/vuepress-site/docs/guides/customize-themes/index.md (hunk 2 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/customize-themes/index.md:99]: Same as above comments.

**Pull request merged**